recipes/opencolorio/all/patches/1.1.1.patch

```diff
@@ -304,8 +189,21 @@ index 4955f4d..14b5017 100644
      ${Boost_INCLUDE_DIR}
      )
  
+diff --git a/src/core/CDLTransform.cpp b/src/core/CDLTransform.cpp
+index 8b05debc..37f8d1cd 100644
+--- a/src/core/CDLTransform.cpp
++++ b/src/core/CDLTransform.cpp
+@@ -126,7 +126,7 @@ OCIO_NAMESPACE_ENTER
+             TiXmlPrinter printer;
+             printer.SetStreamPrinting();
+             doc.Accept( &printer );
+-            return printer.Str();
++            return printer.CStr();
```

Comment:
Constructing the `BuildXML` result `std::string` from `const char*` returned from `printer.CStr()` seems a little less efficient, but I guess it's the only way?

Reply:
Yep. Not perfect for sure, but it was the smallest change I could come up with. I guess I could in theory do `#ifdef TIXML_USE_STL` block to handle STL/non-STL scenarios if that would be desired. Let me know if I should add that.

Reply:
Actually, when `TIXML_USE_STL` is not defined, your patch is the only solution for OCIO, I think.

But what happens if a user specified `tinyxml:with_stl=True`? Then this one-line patch should not be applied?
If so, maybe either the recipe should raise invalid config if  `tinyxml:with_stl=True`, or the patch should be applied conditionally?

Reply:
With your new patch, does it work locally with and without overriding `with_stl` by client?

Reply:
I have now added the following block instead:

    #ifdef TIXML_USE_STL
        return printer.Str();
    #else
        return printer.CStr();
    #endif

I think that this should be optimal solution for both STL and non-STL versions of tinyxml and compile fine in all cases.

Reply:
if it uses STL, shouldn't it be a `printer.c_str()` or maybe `printer.data()`?

Reply:
Agreed